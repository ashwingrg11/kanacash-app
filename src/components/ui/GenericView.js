import React from 'react';
import {
  ScrollView,
  View,
  StyleSheet,
  SafeAreaView,
  TouchableWithoutFeedback,
  Keyboard,
} from 'react-native';
import PropTypes from 'prop-types';
import theme from '~/components/theme/Style';
// import {showLoader, hideLoader} from '~/store/actions/LoaderAction';
// import {useDispatch} from 'react-redux';

const DismissKeyboardHOC = Comp => {
  return ({children, disabled, ...props}) => (
    <TouchableWithoutFeedback
      onPress={Keyboard.dismiss}
      accessible={false}
      disabled={disabled}>
      <Comp {...props}>{children}</Comp>
    </TouchableWithoutFeedback>
  );
};

const DismissKeyboardView = DismissKeyboardHOC(View);

const GenericTemplate = ({
  children,
  footer,
  scrollable,
  style,
  padding,
  bounces,
  attributes,
  backgroundColor,
  header,
  // loading,
  keyboardView,
}) => {
  // const dispatch = useDispatch();
  const ViewGroup = scrollable ? ScrollView : View;

  // React.useEffect(() => {
  //   if (loading) {
  //     dispatch(showLoader());
  //   } else {
  //     dispatch(hideLoader());
  //   }
  // }, [loading]);

  return (
    <SafeAreaView style={[styles.container(backgroundColor)]}>
      {header}
      <View style={styles.body(padding)}>
        <ViewGroup
          bounces={bounces}
          style={[styles.content(backgroundColor, padding), style]}
          keyboardShouldPersistTaps="always"
          {...attributes}>
          <DismissKeyboardView style={styles.full} disabled={!keyboardView}>
            {children}
          </DismissKeyboardView>
        </ViewGroup>
      </View>
      {footer}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  full: {
    flex: 1,
  },
  container: backgroundColor => ({
    flex: 1,
    backgroundColor: backgroundColor,
  }),
  body: padding => ({
    flex: 1,
    paddingTop: padding ? theme.themeMainPadding : 0,
  }),
  content: (backgroundColor, padding) => ({
    flex: 1,
    backgroundColor: backgroundColor,
    paddingHorizontal: padding ? theme.themeMainPadding : 0,
  }),
});

GenericTemplate.propTypes = {
  header: PropTypes.element,
  children: PropTypes.any.isRequired,
  scrollable: PropTypes.bool,
  footer: PropTypes.element,
  style: PropTypes.object,
  padding: PropTypes.bool,
  // loading: PropTypes.bool,
  keyboardView: PropTypes.bool,
};

GenericTemplate.defaultProps = {
  keyboardView: false,
  // loading: false,
  backgroundColor: theme.backgroundColor,
  bounces: true,
  padding: false,
  scrollable: false,
  header: <></>,
  footer: <></>,
  style: {},
};

export default GenericTemplate;
