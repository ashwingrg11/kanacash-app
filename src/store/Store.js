import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';
import {composeWithDevTools} from 'redux-devtools-extension';
import AsyncStorage from '@react-native-community/async-storage';

const config = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
  blackList: [],
};

const reducer = persistReducer(config, rootReducer);
const middleware = composeWithDevTools(applyMiddleware(thunk));

export default data => {
  const store = createStore(reducer, middleware);
  const persistor = persistStore(store);
  return {persistor, store};
};
