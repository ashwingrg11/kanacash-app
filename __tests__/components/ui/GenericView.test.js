import React from 'react';
import {SafeAreaProvider, View} from 'react-native-safe-area-context';
import GenericView from '~/components/ui/GenericView';
import Store from '~/store/Store';

import renderer from 'react-test-renderer';

describe('<GenericView/>', () => {
  it('GenericView renders correctly', () => {
    const tree = renderer
      .create(
        <SafeAreaProvider store={Store}>
          <GenericView><View/></GenericView>,
        </SafeAreaProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
