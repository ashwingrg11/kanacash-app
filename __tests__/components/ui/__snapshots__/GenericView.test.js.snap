// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericView/> GenericView renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
